File: Linked-Lists/C++/linked list traversal.cpp
#include<bits/stdc++.h>
using namespace std;
class Node{
    public:
    int data;
    Node* next;
    public:
    Node(int data1,Node* next1)
    {
        data=data1;
        next=next1;
    }
    public:
    Node(int data1)
    {
        data=data1;
        next=nullptr;
    }
};
Node* convertarraytolist(vector<int> &arr)
    {
    	Node* head=new Node(arr[0]);
    	Node* mover=head;
    	for(int i=1;i<arr.size();i++)
    	{
    		Node* temp=new Node(arr[i]);
    		mover->next=temp;
    		mover=temp;
		}
		return head;
	}
int main()
{
	vector<int>arr={15,5,8,7,2,1};
    Node* head=convertarraytolist(arr);
    
    Node* temp=head;
    while(temp)
    {
    	cout<<temp->data<<" ";
    	temp=temp->next;
	}

}

File: Linked-Lists/Java/LinkedList_Traversal.java
class Node1{
    public
    int data;
    Node next;
    public
    Node1(int data1,Node next1)
    {
        data=data1;
        next=next1;
    }
    public
    Node1(int data1)
    {
        data=data1;
        next=null;
    }
};
public class LinkedList_Traversal
{
    public static Node convertarraytolist(int[]  arr)
{
    Node head=new Node(arr[0]);
    Node mover=head;
    for(int i=1;i<arr.length;i++)
    {
        Node temp=new Node(arr[i]);
        mover.next=temp;
        mover=temp;
    }
    return head;
}
    public static void main(String[] args) {
        int[] arr={15,5,8,7,2,1};
        Node head=convertarraytolist(arr);

        Node temp=head;
        while(temp!=null)
        {
            System.out.print(temp.data+" ");
            temp=temp.next;
        }

    }
}